lib/PuppeteerSharp.Tests/Page/SetExtraHttpHeadersTests.cs

```diff
@@ -15,8 +15,10 @@ public class SetExtraHttpHeadersTests : PuppeteerPageBaseTest
                 ["Foo"] = "Bar"
             });
 
-            var response = await Page.GoToAsync($"{TestConstants.ServerUrl}/headertests/test");
-            Assert.Equal("Bar", await response.TextAsync());
+            var headerTask = Server.WaitForRequest("/empty.html", request => request.Headers["Foo"]);
+            await Task.WhenAll(Page.GoToAsync(TestConstants.EmptyPage), headerTask);
+
+            Assert.Equal("Bar", await headerTask);
```

Comment:
I think `headerTask.Result` it's more clear here. The task was already awaited on 19